### PR TITLE
Send sms service

### PR DESCRIPTION
**Pull request by samodovdi**

### CODE REVIEW

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        [HttpPost]
+        [Route("twilio")]
+        public async Task<IActionResult> TwilioCallback([FromForm] TwilioCallbackModel model)
+        {
+            if (model == null)
+                return Ok();
+            
+            _log.WriteInfo(nameof(TwilioCallback), model, "Twilio callback");
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: we have this log on error
_log.WriteWarning(nameof(SendSmsAsync), new
.Sid,
= response.To.SanitizePhone(),
.Status,
.ErrorCode,
.ErrorMessage
}, "twilio error response");

in context we have property Phone
here log about success responce, will be good add in context property Phone
abd after we can simple make visualization in kibana - success\error by phone codes

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        [HttpPost]
+        [HttpGet]
+        [Route("nexmo")]
+        public async Task<IActionResult> NexmoCallback(NexmoCallbackModel model)
+        {
+            if (model == null)
+                return Ok();
+            
+            _log.WriteInfo(nameof(NexmoCallback), model, "Nexmo callback");
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: we have this log on error
_log.WriteWarning(nameof(SendSmsAsync), new
.Sid,
= response.To.SanitizePhone(),
.Status,
.ErrorCode,
.ErrorMessage
}, "twilio error response");

in context we have property Phone
here log about success responce, will be good add in context property Phone
abd after we can simple make visualization in kibana - success\error by phone codes

File: src/Lykke.Service.SmsSender/Sagas/Commands/ProcessSmsCommand.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using ProtoBuf;
+
+namespace Lykke.Service.SmsSender.Sagas.Commands
+{
+    [ProtoContract]
+    public class ProcessSmsCommand
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender/Sagas/Commands/ProcessSmsCommand.cs]: we can add requariment here - send ClientId?
will be good in log write info about clientId, becouse if we have problem with client, we can found this event by ClientId

File: src/Lykke.Service.SmsSender/Sagas/Commands/SendSmsCommand.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using Lykke.Service.SmsSender.Core.Domain;
+using ProtoBuf;
+
+namespace Lykke.Service.SmsSender.Sagas.Commands
+{
+    [ProtoContract]
+    public class SendSmsCommand
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender/Sagas/Commands/SendSmsCommand.cs]: we can add requariment here - send ClientId?
will be good in log write info about clientId, becouse if we have problem with client, we can found this event by ClientId

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+        [HttpPost]
+        [Route("twilio")]
+        public async Task<IActionResult> TwilioCallback([FromForm] TwilioCallbackModel model)
+        {
+            if (model == null)
+                return Ok();
+            
+            _log.WriteInfo(nameof(TwilioCallback), model, "Twilio callback");
+
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: and will be good add in context CountryCode

File: src/Lykke.Service.SmsSender.Services/SmsSenders/Twilio/TwilioSmsSender.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+        public async Task<string> SendSmsAsync(string phone, string message, string countryCode)
+        {
+            try
+            {
+                var response = await $"{BaseApiUrl}/Accounts/{_settings.ApiKey}/Messages.json"
+                    .WithBasicAuth(_settings.ApiKey, _settings.ApiSecret)
+                    .PostUrlEncodedAsync(new
+                    {
+                        To = phone,
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender.Services/SmsSenders/Twilio/TwilioSmsSender.cs]: and will be good add in context CountryCode

File: src/Lykke.Service.SmsSender.AzureRepositories/SmsRepository/SmsMessageEntity.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        public string Id { get; set; }
+        public string MessageId { get; set; }
+        public string Phone { get; set; }
+        public string Message { get; set; }
+        public SmsProvider Provider { get; set; }
+        public string CountryCode { get; set; }
+        public DateTime Created { get; set; }
+
+        internal static string GenerateRowKey(string id) => id;
+        internal static string GeneratePartitionKey() => "SmsMessage";
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender.AzureRepositories/SmsRepository/SmsMessageEntity.cs]: Maybe split messages by date. Here we can have one big partition

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs (hunk 1 above)
Comment (samodovdi) [reply src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: we have logging about successful delivery in the SmsCommandHandler, on SmsDeliveredCommand processing, but I will add sanitizing for the callback request itself

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs (hunk 2 above)
Comment (samodovdi) [reply src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: the same as for Twilio

File: src/Lykke.Service.SmsSender/Sagas/Commands/ProcessSmsCommand.cs (hunk 3 above)
Comment (samodovdi) [reply src/Lykke.Service.SmsSender/Sagas/Commands/ProcessSmsCommand.cs]: currently, we don't have clientId passed to the service. Need to change the other services for this. Can be implemented in a separate task

File: src/Lykke.Service.SmsSender/Sagas/Commands/SendSmsCommand.cs (hunk 4 above)
Comment (samodovdi) [reply src/Lykke.Service.SmsSender/Sagas/Commands/SendSmsCommand.cs]: the same for the clientId

File: src/Lykke.Service.SmsSender/Controllers/CallbackController.cs (hunk 5 above)
Comment (samodovdi) [reply src/Lykke.Service.SmsSender/Controllers/CallbackController.cs]: it's just logging of the callback request itself, the country code is logged in the SmsCommandHandler

File: client/Lykke.Service.SmsSender.Client/AutorestClientMapper.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Lykke.Service.SmsSender.Client
+{
+    public static class AutorestClientMapper
```
Comment (NovichikhinAlexey) [inline client/Lykke.Service.SmsSender.Client/AutorestClientMapper.cs]: maybe this file not need

File: src/Lykke.Service.SmsSender.Services/SmsSenders/Nexmo/NexmoSmsSender.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+using Lykke.Service.SmsSender.Core.Settings.ServiceSettings.SenderSettings;
+
+namespace Lykke.Service.SmsSender.Services.SmsSenders.Nexmo
+{
+    public class NexmoSmsSender : ISmsSender
+    {
+        private readonly string _baseUrl;
+        private readonly ProviderSettings _settings;
+        private readonly ILog _log;
+        private const string BaseUrl = "https://rest.nexmo.com";
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender.Services/SmsSenders/Nexmo/NexmoSmsSender.cs]: maybe this save in settings?

File: src/Lykke.Service.SmsSender.Services/SmsSenders/Twilio/TwilioSmsSender.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+using Lykke.Service.SmsSender.Core.Settings.ServiceSettings.SenderSettings;
+
+namespace Lykke.Service.SmsSender.Services.SmsSenders.Twilio
+{
+    public class TwilioSmsSender : ISmsSender
+    {
+        private readonly string _baseUrl;
+        private readonly ProviderSettings _settings;
+        private readonly ILog _log;
+        private const string BaseApiUrl = "https://api.twilio.com/2010-04-01";
```
Comment (NovichikhinAlexey) [inline src/Lykke.Service.SmsSender.Services/SmsSenders/Twilio/TwilioSmsSender.cs]: maybe this save in settings?

**Pull request merged**